src/net/tcp.rs

```diff
@@ -59,50 +57,21 @@ impl TcpListener {
     /// future's task. It's recommended to only call this from the
     /// implementation of a `Future::poll`, if necessary.
     pub fn accept(&mut self) -> io::Result<(TcpStream, SocketAddr)> {
-        loop {
-            if let Some(mut pending) = self.pending_accept.take() {
-                match pending.poll().expect("shouldn't be canceled") {
-                    Async::NotReady => {
-                        self.pending_accept = Some(pending);
-                        return Err(io::ErrorKind::WouldBlock.into())
-                    },
-                    Async::Ready(r) => return r,
-                }
-            }
-
-            if let Async::NotReady = self.io.poll_read() {
-                return Err(io::Error::new(io::ErrorKind::WouldBlock, "not ready"))
-            }
+        if let Async::NotReady = self.io.poll_read() {
+            return Err(io::Error::new(io::ErrorKind::WouldBlock, "not ready"))
```

Comment:
Wouldn't `Err(io::ErrorKind::WouldBlock.into())` avoid an allocation here? I don't think "not ready" add much to the error message.

Reply:
Indeed! This PR was mostly just moving things around not actually creating new instances of this. Feel free to send a PR to update this!

Reply:
Done https://github.com/tokio-rs/tokio/pull/65.